### PR TITLE
add: type definition for pino-datadog

### PR DESCRIPTION
**Pull request by czystyl**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/pino-datadog/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// Type definitions for pino-datadog 2.0
+// Project: https://github.com/ovhemert/pino-datadog
+// Definitions by: czystyl <https://github.com/me>
```
Comment (peterblazejewicz) [inline types/pino-datadog/index.d.ts:3]:
```suggestion
// Definitions by: czystyl <https://github.com/czystyl>
```

**Review by peterblazejewicz — COMMENTED** [review]
alll the options are optional:
https://github.com/ovhemert/pino-datadog/blob/master/src/index.js
and all props are optional I think, when there is no options,
default `size` is provided

File: types/pino-datadog/index.d.ts (hunk 1 above)
Comment (czystyl) [reply types/pino-datadog/index.d.ts:3]: Thanks!

**Comment by czystyl** [comment]
> alll the options are optional:
> [ovhemert/pino-datadog@`master`/src/index.js](https://github.com/ovhemert/pino-datadog/blob/master/src/index.js)
> and all props are optional I think, when there is no options,
> default `size` is provided

Oh looking from this perspective - Yes. The library is a connection to Datadog and when apiKey is not provided the library will throw an error each time. This is why I put apiKey as a required key in the options.

Also looking into documentation, the author had in mind to make this required, see here: ovhemert/pino-datadog@master/docs/API.md#apikey

Thanks for the review!

**Comment by peterblazejewicz** [comment]
OK, so what's the most common scenario? (for us, users)
I understand api key could be required, so make it required (property). options do seem to be optional by design:
https://github.com/ovhemert/pino-datadog/blob/master/test/index.test.js
but we can go a way you want

**Comment by czystyl** [comment]
> OK, so what's the most common scenario? (for us, users)
> I understand api key could be required, so make it required (property). options do seem to be optional by design:
> [https://github.com/ovhemert/pino-datadog/blob/master/test/index.test.js](https://github.com/ovhemert/pino-datadog/blob/master/test/index.test.js?rgh-link-date=2021-09-09T19%3A42%3A05Z)
> but we can go a way you want

I see, and you have right that from the code perspective having it optional should be a way to go. From the user perspective, the library without specifying ApiKey is (IMHO) useless because generates only errors. 

So I thought to be in line with the documentation. Another thing that I had in mind is TypeScirpt should help catch errors on compile time. 

Those are my thought but also I'm not much experienced in writing DefinitelyTyped so if we should be 100% in line with the code then I can change it as well.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@czystyl thx!
we went with proposed implementation that requires api key usage. Feel free to amend with future PRs.
x-ref: ovhemert/pino-datadog#62

**Comment by czystyl** [comment]
Ready to merge

**Pull request merged**